api/v1beta1/gcpcluster_webhook.go

```diff
@@ -33,36 +35,44 @@ var clusterlog = logf.Log.WithName("gcpcluster-resource")
 
 // SetupWebhookWithManager sets up and registers the webhook with the manager.
 func (c *GCPCluster) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	w := new(gcpClusterWebhook)
```

Comment:
Could we move the webhooks to `internal/webhooks` so that we start brining the structure in line with upstream CAPI and also kubebuilder

Reply:
This sounds reasonable. But can this be done in a separate pr (preferrably by project maintainers), as this will look like a lot of change in the pr and mask the actual change needed to make things work with the kubebuilder api breakage?

Reply:
Yeah +1 for doing this, but probably best to do it in a follow up PR to avoid polluting this one